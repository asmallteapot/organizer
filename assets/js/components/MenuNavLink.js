import React from 'react'
import { Route, Link } from 'react-router-dom'

export default ({
  to,
  exact,
  strict,
  location,
  activeClassName,
  className,
  activeStyle,
  style,
  isActive: getIsActive,
  children
}) => (
  <Route
    path={to}
    exact={exact}
    strict={strict}
    location={location}
    children={({ location, match }) => {
      const isActive = !!(getIsActive ? getIsActive(match, location) : match)

      return (
        <li className={isActive ? [className, (activeClassName ? activeClassName : "active")].filter(i => i).join(' ') : className}>
          <Link to={to}>{children}</Link>
        </li>
      )
    }}
  />
);


