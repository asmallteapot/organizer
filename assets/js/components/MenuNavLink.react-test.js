import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter as Router }  from 'react-router-dom'
import MenuNavLink from './MenuNavLink'

const WrappedLink = (props) => (
  <Router>
    <MenuNavLink {...props} />
  </Router>
)

test('Smoke test', () => {
  const component = mount(<WrappedLink to="/" />);
  expect(component.html()).toMatchSnapshot();
});
