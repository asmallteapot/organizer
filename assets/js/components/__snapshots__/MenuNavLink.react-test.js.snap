// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke test 1`] = `"<li class=\\"active\\"><a href=\\"/\\"></a></li>"`;
